workflow/artifacts/hdfs/hdfs.go

```diff
@@ -73,6 +80,14 @@ func ValidateArtifact(errPrefix string, art *wfv1.HDFSArtifact) error {
 	if hasKrbCCache && (art.KrbServicePrincipalName == "" || art.KrbConfigConfigMap == nil) {
 		return errors.Errorf(errors.CodeBadRequest, "%s.krbServicePrincipalName and %s.krbConfigConfigMap are required with %s.krbCCacheSecret", errPrefix, errPrefix, errPrefix)
 	}
+
+	switch art.DataTransferProtection {
+	case "", DataTransferProtectionAuthentication, DataTransferProtectionIntegrity, DataTransferProtectionPrivacy:
+		// Do nothing
+	default:
+		return errors.Errorf(errors.CodeBadRequest, "%s.dataTransferProtection must be one of 'authentication', 'integrity', 'privacy', or empty", errPrefix)
+	}
```

Comment:
Thanks for adding the validation here, although I'm torn on whether we should validate this or not. On one hand, if the `ClientOptions` ever change for `DataTransferProtection` then we'd have to update this when the version in go.mod gets updated.

Reply:
I agree that your concern is valid. I initially added validation based on the three modes specified in https://github.com/colinmarc/hdfs/blob/master/client.go#L24-L28. While I don't expect these options to change frequently, **I agree that if they do change in the future, this validation might be easily overlooked.**